src/lua/styles.c

```diff
@@ -259,8 +259,14 @@ int dt_lua_style_apply(lua_State *L)
     luaA_to(L, dt_style_t, &style, 1);
     luaA_to(L, dt_lua_image_t, &imgid, 2);
   }
-  dt_styles_apply_to_image(style.name, FALSE, FALSE, imgid);
-  DT_DEBUG_CONTROL_SIGNAL_RAISE(darktable.signals, DT_SIGNAL_TAG_CHANGED);
+
+  if(darktable.develop && darktable.develop->image_storage.id == imgid)
+    dt_print(DT_DEBUG_LUA, "LUA ERROR : %s can't called from darkroom view.  Use darktable.gui.action() instead\n", __FUNCTION__);
```

Comment:
Have you tried calling:

`    dt_styles_apply_to_dev(style.name, darktable.develop->image_storage.id);`

When in darkroom?

Reply:
Repeatedly with lots of crashes as the result.  The crashes are segfaults, usually updating module instance names, but sometimes updating the history, and sometimes other places.  I couldn't get anything that repeated consistently.

Reply:
I also tried using dt_action_process(), since darktable.gui.action() works fine from Lua,  but didn't have any better luck.

Reply:
Here's the test script and styles I used if you'd like to try it
[styles_apply.zip](https://github.com/darktable-org/darktable/files/11240419/styles_apply.zip)

Reply:
Where should I put all those files and what's the procedure to trigger the issue? I'm not using Lua :)

Also, just to be clear I meant to call the `dt_styles_apply_to_dev() `only in the conditional that you have added:
`
if(darktable.develop && darktable.develop->image_storage.id == imgid)`